autoblocks/_impl/tracer.py

```diff
@@ -238,13 +214,31 @@ async def _send_event_unsafe(
             except Exception as err:
                 log.error("Unable to evaluate events. Error: %s", err, exc_info=True)
 
-        req = await global_state.http_client().post(
-            url=INGESTION_ENDPOINT,
-            json=payload.to_json(),
-            headers=self._client_headers,
-            timeout=self._timeout_seconds,
-        )
-        req.raise_for_status()
+        try:
+            if global_state.main_thread_has_finished():
```

Comment:
I would just add one comment about how this is an edge case for normal shutdown

Reply:
it will happen for both plain script and sigterm because we are not starting the flush until the main thread has joined